lang/rust/avro/src/schema.rs

```diff
@@ -241,17 +242,41 @@ impl Name {
         }
     }
 
+    /// Create a new `Name`.
+    /// Parses the optional `namespace` from the `name` string.
+    /// `aliases` will not be defined.
+    pub fn parse_str(name: &str) -> Name {
```

Comment:
I think It would make sense to run this on the input to `Name::new`. I think it would make the SDK more consistent. I would be worried by confusing of `Name::new(&schema_str[..]) != Name::parse_str(&schema_str[..])`

Reply:
If we are worried about breaking backwards compatibility for those who were using "namespace.name", might be work deprecating "new" in favor of this? Or an implementation of `From<&str>` for Name?

Reply:
I also wondered whether to modify `new()` but its documentation explicitly says that it doesn't parse and set the namespace.
Since you also noticed it I think we should just do it, because currently `new()` is broken!

Reply:
Done!